File: Pen&Paper/Paper.cpp
#include "Paper.hpp"

Paper::Paper(int maxSymbols=4096) {
    this->maxSysmbols = maxSysmbols;
}

Paper::~Paper() {}

int Paper::getMaxSymbols() const {
    return this->maxSysmbols;
}

int getSymbols() const {
    return this->symbols;
}

void Paper::addContent(const std::string& message) {
    if ( this->maxSysmbols <= this->content.length() ) {
        throw OutOfSpaceException();
    }
    int available = this->maxSysmbols - this->content.length();
    if ( message.length() > available ) {
        this->content += message.substr(0, available);
        return;
    }
    this->content += message;
}

void show() const {
    return this->content;
}

// std::ostream& operator<<(std::ostream& out, const Paper paper) {
//     out << "Paper(" << paper.getMaxSymbols() << "/";
//     out << paper.getContent().length() << ")";
//     return out;
// }

File: Java/Pen&Paper/Paper.java
public class Paper {
    private StringBuffer content;
    private int maxSymbols;
    private int symbols;

    public Paper(int maxSymbols) {
        this.maxSymbols = maxSymbols;
        this.symbols = 0;
        content = new StringBuffer();
    }

    public Paper() {
        this.maxSymbols = 4096;
        this.symbols = 0;
        content = new StringBuffer();
    }

    public int getMaxSymbols() {
        return this.maxSymbols;
    }

    public int getSymbols() {
        return this.symbols;
    }

    public void addContent(String message) throws OutOfSpaceException {
        int total = content.length() + message.length();

        if ( content.length() == maxSymbols ) {
            throw new OutOfSpaceException();
        }
        if ( total > maxSymbols ) {
            content.append(message.substring(0, maxSymbols-content.length()));
            return;
        }
        content.append(message);
    }
    public void show() {
        System.out.println(content.toString());
    }
}